packages/integration-sdk-runtime/src/execution/__tests__/executeIntegration.test.ts

```diff
@@ -341,6 +341,164 @@ describe('executeIntegrationInstance', () => {
     });
   });
 
+  test('includes types for partial datasets declared in subsequent step meta data when dependency failure', async () => {
+    const config = createInstanceConfiguration({
+      invocationConfig: {
+        integrationSteps: [
+          {
+            id: 'my-step-a',
+            name: 'My awesome step',
+            entities: [
+              {
+                resourceName: 'The Test',
+                _type: 'test_a',
+                _class: 'Test',
+              },
+            ],
+            relationships: [],
+            executionHandler: jest
+              .fn()
+              .mockRejectedValue(new Error('something broke')),
+          },
+          {
+            id: 'my-step-b',
+            name: 'My awesome step',
+            entities: [
+              {
+                resourceName: 'The Test',
+                _type: 'test_b',
+                _class: 'Test',
+              },
+            ],
+            relationships: [],
+            dependsOn: ['my-step-a'],
+            executionHandler: jest.fn(),
+            partialDatasets: [
+              {
+                types: ['test_b2'],
+              },
+            ],
+          },
+        ],
+      },
+    });
+
+    await expect(
+      executeIntegrationInstance(
+        config.logger,
+        config.instance,
+        config.invocationConfig,
+      ),
+    ).resolves.toEqual({
+      integrationStepResults: [
+        {
+          id: 'my-step-a',
+          name: 'My awesome step',
+          declaredTypes: ['test_a'],
+          encounteredTypes: [],
+          status: StepResultStatus.FAILURE,
+        },
+        {
+          id: 'my-step-b',
+          name: 'My awesome step',
+          declaredTypes: ['test_b'],
+          encounteredTypes: [],
+          dependsOn: ['my-step-a'],
+          partialDatasets: [
+            {
+              types: ['test_b2'],
+            },
+          ],
+          status: StepResultStatus.PARTIAL_SUCCESS_DUE_TO_DEPENDENCY_FAILURE,
+        },
+      ],
+      metadata: {
+        partialDatasets: {
+          types: ['test_a', 'test_b2', 'test_b'],
```

Comment:
I don't think you have a test about this but I just want to be clear - if `step-a` has a partial dataset, but is successful, it still has `StepResultStatus.SUCCESS`, correct? and `step-b`, which depends on `step-a`, will be marked as `StepResultStatus.SUCCESS` as opposed to `StepResultStatus. PARTIAL_SUCCESS_DUE_TO_DEPENDENCY_FAILURE`?

Reply:
Def would be good to add a test for the `SUCCESS` case.

Reply:
That's correct @ndowmon, added a test 👍